database/rrdcalc.c

```diff
@@ -666,10 +666,16 @@ static void rrdcalc_labels_unlink_alarm_loop(RRDHOST *host, RRDCALC *alarms) {
                  host->hostname,
                  rc->host_labels);
 
+            RRDCALC *rc_old = rc->next;
```

Comment:
It looks like you are skipping rrdcalc structures - rc->next is assigned twice.

Reply:
Maybe it's better to rename `rc_old`->`rc_next` and remove `rc = rc->next` from `for(;;)`.

Reply:
Thank you @vlvkobal , indeed one rc was skipped after freeing one.